xbmc/filesystem/FileCache.cpp

```diff
@@ -268,23 +272,31 @@ void CFileCache::Process()
           sourceSeekFailed = true;
         }
       }
-      if (!sourceSeekFailed)
+
+      if (sourceSeekFailed)
       {
-        const bool bCompleteReset = m_pCache->Reset(m_seekPos, false);
+        // Seek failed: Reset cache to current source position
+        bCompleteReset = m_pCache->Reset(m_source.GetPosition(), false);
+      }
+      else
+      {
+        // Seek done: Update variables
         m_readPos = m_seekPos;
-        m_writePos = m_pCache->CachedDataEndPos();
-        assert(m_writePos == cacheMaxPos);
-        average.Reset(m_writePos, bCompleteReset); // Can only recalculate new average from scratch after a full reset (empty cache)
-        limiter.Reset(m_writePos);
         m_nSeekResult = m_seekPos;
-        if (bCompleteReset)
-        {
-          CLog::Log(LOGDEBUG,
-                    "CFileCache::{} - <{}> cache completely reset for seek to position {}",
-                    __FUNCTION__, m_sourcePath, m_seekPos);
-          m_bFilling = true;
-          m_bLowSpeedDetected = false;
-        }
+      }
+
+      m_writePos = m_pCache->CachedDataEndPos();
+      assert(m_writePos == cacheMaxPos);
+      average.Reset(
+          m_writePos,
+          bCompleteReset); // Can only recalculate new average from scratch after a full reset (empty cache)
```

Comment:
the formatting would probably look better if you move the comment to it's own line

Reply:
Done